lib/Echidna/Test.hs

```diff
@@ -66,11 +68,20 @@ checkETest t = do
         Nothing -> False
   res <- case t of
     -- If our test is a regular user-defined test, we exec it and check the result
-    Left  (f, a) -> execTx (Tx (SolCall (f, [])) (s a) a g 0 0 (0, 0)) >> gets (p f . getter)
+    Left  (f, a) -> do
+      sd <- hasSelfdestructed a
+      _  <- execTx (Tx (SolCall (f, [])) (s a) a g 0 0 (0, 0)) 
+      b  <- gets (p f . getter)
+      return $ not sd && b
     -- If our test is an auto-generated assertion test, we check if we failed an assert on that fn
-    Right sig    -> (||) <$> fmap matchR       (use $ hasLens . result)
-                         <*> fmap (matchC sig) (use $ hasLens . state . calldata . _1)
-  put vm -- restore EVM state
+    Right sig    -> do
+      vm' <- use hasLens
+      b1 <- fmap matchR       (use $ hasLens . result)
```

Comment:
we should rename these conditions, which were renamed in `master`

Reply:
Fixed.